### PR TITLE
refactor: ScreenShare event to work with the new analytics mechanism

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

In this PR I transition the screenShare event to work with the new mechanism. In addition to that I fixed an issue regarding a failing test from the previous PR's regarding refactoring analytics. We were calling the `tagEvent` method from the previous implementation before the refactoring. Now we're calling the correct (updated) `tagEvent` method and the failing test has been fixed.

### Notes
Remaining stuff to be done is to transition the `EndCall` event to work with the new mechanism and then clean things up by removing old code etc.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift [hunk 1]
```diff
@@ -165,12 +165,10 @@ extension AnalyticsCallingTracker: WireCallCenterCallParticipantObserver {
         } else if let screenSharedParticipant = participants.first(where: { $0.state.videoState == .stopped && ($0.user as? ZMUser != selfUser) }),
                   let screenSharingDate = screenSharingStartTimes[screenSharedParticipant.clientId],
                   let conversationId = conversation.remoteIdentifier,
-                  let callInfo = callInfos[conversationId] {
+                  let _ = callInfos[conversationId] {
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:168]: The call info doesn't seem to be needed so I think it's safe to delete this line.

File: Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift [hunk 2]
```diff
@@ -165,12 +165,10 @@ extension AnalyticsCallingTracker: WireCallCenterCallParticipantObserver {
...
 
             // When videoState == .stopped from a remote participant, tag the event if we found a record in screenSharingInfos set with matching clientId
-            analytics.tag(callEvent: .screenSharing(duration: -screenSharingDate.timeIntervalSinceNow),
-                          in: conversation,
-                          callInfo: callInfo)
+            Analytics.shared.tagEvent(.screenShare(callDirection: .outgoing, duration:  -screenSharingDate.timeIntervalSinceNow, in: conversation))
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:171]: This appears to be screenshares from other participants (not the self user), in which case I would say the the direction would be `incoming`. What do you think?

File: Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift [hunk 3]
```diff
@@ -80,8 +80,7 @@ extension Analytics {
             attributes.merge(attributesForVideoToogle(with: callInfo), strategy: .preferNew)
             attributes.merge(["reason": reason], strategy: .preferNew)
         case .screenSharing(let duration):
-            attributes["screen_share_direction"] = "incoming"
-            attributes["screen_share_duration"] = Int(round(duration / 5)) * 5
+            Analytics.shared.tagEvent(.screenShare(callDirection: .incoming, duration: duration, in: conversation))
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: @David-Henner is the `screenSharing` call event related only to the self user? If so, @agisilaos I think that the direction here should actually be `outgoing`.

File: Wire-iOS/Sources/Analytics/Events/Analytics+CallStateEvents.swift [hunk 4]
```diff
@@ -70,4 +75,9 @@ private extension AnalyticsAttributeKey {
     /// Expected to refer to a value of type `AnalyticsCallDirectionType`.
     static let callDirection  = AnalyticsAttributeKey(rawValue: "call_direction")
 
+    /// The duration of the screen-sharing.
+    ///
+    /// Expected to refer to a value of type `AnalyticsScreenShareDurationType`.
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Analytics/Events/Analytics+CallStateEvents.swift:80]: Do you mean to say?:

"Expected to refer to a value of type `RoundedInt`."

File: Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:171]: Correct, good catch!

File: Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift (hunk 3 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: Looking forward to what @David-Henner has to say.

File: Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift (hunk 1 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:168]: But the comment a bit down the line says:

`// When videoState == .stopped from a remote participant, tag the event if we found a record in screenSharingInfos set with matching clientId` 

So we maybe need the callInfo and if all constants match their appropriate values we call the method to track the event.

Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:168]: If I remove this line then I get the same warning that the `let conversationId = conversation.remoteIdentifier`. Should we remove both of those lines?

Comment (johnxnguyen) [reply Wire-iOS/Sources/Analytics/AnalyticsCallingTracker.swift:168]: Sure, but I think that info wasn't being stored in `callInfo`, but is what `screenSharingDate` is for. The call info was previously being passed into the tag method so some attributes could be recorded, but we don't need those anymore it seems. 

If we get rid of `callInfo`, then we also don't need `conversationId`, so we can delete them both.

File: Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift (hunk 3 above)
Comment (David-Henner) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: I'm not familiar with this but as far as I can see this event is only used in `AnalyticsCallingTracker.swift:171`, which is not at all related to the self user. In that case the direction should be `.incoming`

Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: So in line `83` should we leave it as it is? In line `171` I already changed it to .`incoming`

Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: @johnxnguyen But the direction here initially was `.incoming` That's why I left it as it is.

Comment (David-Henner) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: > So in line 83 should we leave it as it is? In line 171 I already changed it to .incoming.

Yes we can leave it

Comment (johnxnguyen) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: ok, sounds fine to me!

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift (hunk 3 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Analytics/Events/Analytics+CallEvents.swift:83]: Perfect! 🤩

**Review by sb88k — APPROVED** [review]

**Pull request merged**